lib/Runtime/Library/InJavascript/Intl.js

```diff
@@ -608,20 +637,42 @@
         return toReturn;
     }());
 
+    // mappedDefaultLocale will get the default locale and update any deprecated
+    // collation/sort order values it may use
     var mappedDefaultLocale = function () {
-        var parts = platform.builtInRegexMatch(GetDefaultLocale(), /([^_]*)_?(.+)?/);
-        var locale = parts[1];
-        var collation = parts[2];
-        var availableBcpTags = reverseLocaleAcceptingCollationValues[locale];
+        let locale = undefined;
+        let collation = undefined;
+        if (isICU) {
+            // ICU's getDefaultLocale() will return a valid BCP-47/RFC 5646 langtag
+            locale = platform.getDefaultLocale();
+            const match = platform.builtInRegexMatch(locale, /-u(-[^\-][^\-]?-[^\-]+)*(-co-[^\-]+).*/);
+            if (match) {
+                collation = callInstanceFunc(StringInstanceReplace, match[2], "-co-", "");
```

Comment:
Rather than doing another string operation, write the regex so that it captures the part you want, like this: `(-co-([^\-]+))`.

Reply:
Also what if you got a match but there was no matching group where you expected? `-co-` is forced to match, as well as the value, and we're assuming a well-formed locale from a system API, but you never know. Food for thought.

If you end up finding that there might a problem here: maybe add an extra check that the index is on the object before using the returned value as if it were a string (it might be `undefined`) if that group wasn't matched.

Reply:
so, if `match` exists, `match[2]` must exist, since its not an optional capture group. It must be equal to the collation value, because it seems like the first capture group matches only the last instance of itself in the resulting match array (I changed the regex slightly to only capture the actual collation value):

```js
> "en-US-u-co-thai-ca-calendar-so-methingelse-co-phonebk".match(/-u(-[^\-][^\-]?-[^\-]+)*-co-([^\-]+).*/)
[ '-u-nu-thai-ca-calendar-so-methingelse-co-phonebk',
  '-so-methingelse',
  'phonebk',
  index: 5,
  input: 'en-US-u-co-thai-ca-calendar-so-methingelse-co-phonebk' ]
> "en-US-u-nu-thai-co-phonebk-ca-calendar".match(/-u(-[^\-][^\-]?-[^\-]+)*-co-([^\-]+).*/)
[ '-u-nu-thai-co-phonebk-ca-calendar',
  '-nu-thai',
  'phonebk',
  index: 5,
  input: 'en-US-u-nu-thai-co-phonebk-ca-calendar' ]
```

Reply:
SGTM

Reply:
Do you want to remove the -co-phonebk from that string after extracting it? In that case you'll also need to optionally capture any extensions before and after.

Keep in mind that there are other singletons followed by keys like the private use -x i.e. `-x-aa-aavalue`. The likelihood you'd be dealing with that in this scenario is pretty minimal.

Reply:
My new version extracts "phonebk" and then will do `locale = locale.replace(\`-co-${match[2]}\`, ""). We may end up with lang-tag-u at the end if it was the only Unicode extension, but I believe we handle that later?